.github/workflows/ci-gh.yml

```diff
@@ -0,0 +1,57 @@
+name: CI (GPU)
+
+concurrency:
+  group: ci-gpu-on-${{ github.event_name }}-from-${{ github.ref_name }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches-ignore:
+      - gh-pages  # deployment target branch (this workflow should not exist on that branch anyway)
+  pull_request:
+    branches-ignore:
+      - gh-pages  # deployment target branch (this workflow should not exist on that branch anyway)
+
+jobs:
+  build:
+    permissions: read-all
+    runs-on: ubuntu-latest
+    container:
+      options: -u root
+      image: rapidsai/devcontainers:23.06-cpp-cuda11.8-mambaforge-ubuntu22.04
+      env:
+        PYTHONDONTWRITEBYTECODE: 1
+        SCCACHE_REGION: us-east-2
+        SCCACHE_BUCKET: rapids-sccache-devs
+        SCCACHE_S3_KEY_PREFIX: legate-cunumeric-dev
+        VAULT_HOST: https://vault.ops.k8s.rapids.ai
+        GH_TOKEN: "${{ secrets.PERSONAL_ACCESS_TOKEN }}"
+        GITHUB_TOKEN: "${{ secrets.PERSONAL_ACCESS_TOKEN }}"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        
+      - name: Build
+        shell: bash -eo pipefail {0}
+        run: |
+          apt update;
+          DEBIAN_FRONTEND=noninteractive \
+          apt install -y --no-install-recommends \
+              numactl rlwrap;
+          mv scripts/home/coder/.gitconfig /home/coder/.gitconfig;
+          mv scripts/home/coder/.local /home/coder/.local;
+          chmod a+x /home/coder/.local/bin/*;
+          chown -R coder:coder /home/coder/;
+
+          su - coder -Pc "                                   \
+          JOBS=1                                             \
+          CMAKE_BUILD_PARALLEL_LEVEL=1                       \
```

Comment:
Is it because of the limitations of the Github runners? They have 2 cores, right? But the memory may be insufficient I guess.

Reply:
Yes, from my vague recollections of my chat with Paul regarding this it seems the runner fails to send status updates to GH if both cores are pegged at 100% by the build process and GH registers that as a failed build. So using both cores is a no no. Maybe we can use `nice` to make this work with two cores?

Reply:
Well, 1 core should be enough with sccache I guess.